### PR TITLE
Added new features: CI-CD, SendGrid integration, GCP Cloud Shell option 

### PR DESCRIPTION
**Pull request by zack-amirakulov**
Added new features: 
- CI-CD pipeline configuration for Dev and Prod
- SendGrid integration, "send_email" dag
- GCP Cloud Shell option 
- Comprehensive Readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,72 +1,106 @@
+## Apache Airflow 2 for local development and CI/CD
+=======================================================
+
+This project is an easy-to-use development environment for Apache Airflow version 2. It can be run locally on a variety of OS platforms with simple steps for spinning up Airflow. The project can be integrated into a CI/CD DATA pipeline using GCP Cloud Build to build, test, and deploy workflows into GCP Cloud Composer. The project is meant to address several "infrastructure challenges" that Airflow developers experience and allows them to focus on workflow development rather than platform installation/configuration.
```
Comment (eugenemiretsky) [inline README.md:4]: What's DATA pipeline?

Comment (eugenemiretsky) [reply README.md:4]: Can just have a bullet point list of features: container caching, unit tests, integration tests, linter, etc. (basically each step in cloud build lol )

Comment (eugenemiretsky) [reply README.md:4]: Also can say. "The project provides an opinionated CloudBuild CI/CD pipeline for Composer deployment" (Google likes the word opinionated lol )

Comment (eugenemiretsky) [reply README.md:4]: Could also divide it into 3 parts

1. Local Development environment
2. GCP cloud shell development environment 
3. Opinionated CI/CD pipeline

File: README.md [hunk 2]
```diff
@@ -1,72 +1,106 @@
...
+
+  - 2.4.10 &nbsp; On  <strong> Visual Studio Code </strong>, you now see a green WSL indicator in the bottom left corner, click on it and choose <strong> Open Folder in WSL </strong>. Windows will prompt you to select a folder, provide the follwing path to a folder: <strong> ` \\wsl$\ubuntu\home\airflow ` </strong> , and choose the folder with the Airflow code: (<strong> airflo2-local-cicd </strong>)
+
+  - 2.4.11 &nbsp; Open a terminal session in VS code `(Menu Terminal -- New Terminal)` and run the WLS docker installation script:
+
+        chmod +x ./helpers/scripts/docker-wls.sh && sudo ./helpers/scripts/docker-wls.sh
+
+  - 2.4.12 &nbsp; Proceed with the installation and initialization steps (<strong> section #3 and #4 </strong>)
 
 <br/>
```
Comment (eugenemiretsky) [inline README.md:106]: Reaplce "Dependencies to solve before you go" with "Customizing Airflow Settings"

File: helpers/scripts/composer_dag_send_email.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  status=$?
+  n=$(($n+1))
+  sleep $5
+done
+
+# - Unpause the DAG if paused
+gcloud beta composer environments run $1 --location $2 dags unpause -- $3
+
+# - Run the DAG "semd_email". The email contains commit ID.
+gcloud beta composer environments run $1 --location $2 dags trigger -- $3 --conf '{"key":"'${6}'"}'
```
Comment (eugenemiretsky) [inline helpers/scripts/composer_dag_send_email.sh:18]: should we change "key" to "version"? (here and in the dag).

Comment (zack-amirakulov) [reply helpers/scripts/composer_dag_send_email.sh:18]: It only works with "key" and its value for some reaason. Initially tried "Commit ID" but it did not work.

File: README.md (hunk 1 above)
Comment (zack-amirakulov) [reply README.md:4]: - Changed DATA pipeline to CI/CD pipeline

Comment (zack-amirakulov) [reply README.md]: Revised the instructions in accordance with Eugene's feedback

File: helpers/scripts/composer_dag_send_email.sh (hunk 3 above)
Comment (eugenemiretsky) [reply helpers/scripts/composer_dag_send_email.sh]: loool. so weird

Comment (zack-amirakulov) [reply helpers/scripts/composer_dag_send_email.sh]: I am double checking it right now

Comment (zack-amirakulov) [reply helpers/scripts/composer_dag_send_email.sh]: It actually worked when I double-checked it. I've given a proper name to the parameter. Yeah, weird

**Comment by zack-amirakulov** [comment]
Squash-merged into master

**Pull request merged**